pkg/staking/stakingcontract/contract_test.go

```diff
@@ -0,0 +1,539 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stakingcontract_test
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/staking/stakingcontract"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/transaction"
+	transactionMock "github.com/ethersphere/bee/pkg/transaction/mock"
+)
+
+func TestDepositStake(t *testing.T) {
```

Comment:
could test cases reuse same function for `Call`, `Send` and `WaitForReceip` mocks?

Reply:
we can but won't make much difference because the functions return something different in some cases and with curr implementation it feels more readable tho not sure. I followed the postage contract tests for reference.